packages/client/src/comp/Address/Fields.tsx

```diff
@@ -0,0 +1,157 @@
+import React from 'react'
+import Input from 'muicss/lib/react/input'
+import styled from 'styled-components'
+import { BaseInput } from '../util/Input'
+import { Select, Option } from 'muicss/react'
+import { useControlRef } from '../util/ControlRef'
+import { RoundedButton } from '../util/Button'
+import { allStates } from '../../common'
+import { useAppHistory } from '../../lib/path'
+import { AddressInputPartContainer } from './Container'
+import { FetchingDataContainer } from '../../lib/unstated'
+
+interface Props {
+  type: 'initialAddress' | 'separateMailing'
+}
+
+const FlexBox = styled.div`
+  display: flex;
+  flex-direction: row;
+  justify-content: space-between;
+  flex-flow: row wrap;
+  align-items: center;
+  margin: 1em 0;
+`
+
+const Flex = styled.div<{ basis?: string }>`
+  flex-basis: ${p => p.basis ?? '100%'};
+  /* Fixes needless scrollbars showing on Linux/Windows */
+  .mui-select label { overflow: hidden; }
+`
+
+// Since react-hooks/exhaustive-deps will trigger re-renders due to its
+// depedencies, we declare this function outside of our component.
+const useDidMount = (fun: () => void) => React.useEffect(fun, [])
+
+export const AddressFields: React.FC<Props> = ({ type }) => {
+  const {
+    fields: baseAddressFields,
+    mailingFields,
+    setField: setBaseAddressField,
+    setMailingField,
+  } = AddressInputPartContainer.useContainer()
+  const { path } = useAppHistory()
+  const addrRef = useControlRef<Input>()
+  const fields = type === 'initialAddress' ? baseAddressFields : mailingFields
+  const setField = type === 'initialAddress' ? setBaseAddressField : setMailingField
+  const { fetchingData } = FetchingDataContainer.useContainer()
+  // When we first arrive at page, set focus and move cursor to beginning
+  useDidMount(() => {
+      if (type === 'initialAddress') {
+        if (path?.type === 'address' && addrRef?.current) {
+          // eslint-disable-next-line @typescript-eslint/no-explicit-any
+          const controlEl = (addrRef.current as any).controlEl as HTMLInputElement
+          controlEl.focus({preventScroll: true})
+          controlEl.setSelectionRange(0, 0)
+        }
+      }
+  })
+
+  return <FlexBox>
+    {/* We use a 7:3 basis proportion between Flex Components */}
+    {/* The spacing between them 2% of the basis */}
+
+    <Flex basis='70%'>
```

Comment:
![image](https://user-images.githubusercontent.com/1906321/90565803-535bfe00-e15c-11ea-810b-17a0ee1bda3c.png)

This is al little crowded.  In mobile, let's make this 4 rows:

street
apartment
city
state / zip